RFCs/11-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,728 @@
+
+#### 3.1  Examples
+
+In the following examples, `films` table has a closed schema.
+
+Insert a single row:
+
+```SQL
+INSERT INTO films VALUES
+    ('UA502', 'Bananas', 105, '1971-07-13', 'Comedy', '82 minutes');
+```
+
+In this example, the `len` attribute is omitted in the explicit list and therefore it will have the default value:
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind)
+    VALUES ('T_601', 'Yojimbo', 106, '1961-06-16', 'Drama');
+```
+
+The following examples use the DEFAULT clause for the date attributes rather than specifying a value:
+
+```SQL
+INSERT INTO films VALUES
+    ('UA502', 'Bananas', 105, DEFAULT, 'Comedy', '82 minutes');
+INSERT INTO films (code, title, did, date_prod, kind)
+    VALUES ('T_601', 'Yojimbo', 106, DEFAULT, 'Drama');
+```
+
+The following examples statement inserts a row or item consisting entirely of default values:
+
+```SQL
+INSERT INTO films DEFAULT VALUES;
+```
+
+The following example statement inserts multiple rows using the multirow VALUES syntax:
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind) VALUES
+    ('B6717', 'Tampopo', 110, '1985-02-10', 'Comedy'),
+    ('HG120', 'The Dinner Game', 140, DEFAULT, 'Comedy');
+```
+
+In the following statement, we insert two values to `Music` table, `Music` table has an open schema with `Artist` and `SongTitle` as required attributes:
+
+```SQL
+INSERT INTO Music
+    VALUES
+        <<{'Artist' : 'Acme Band','SongTitle' : 'PartiQL Rocks'},
+        {'Artist' : 'Emca Band','TitleSong' : 'PartiQL Rocks'};>>
+```
+
+In the following statement we Insert multiple person items as a bag value to `Person` table, `Person` table has an open schema with `LastName` and `DOB` as required attributes:
+
+```SQL
+INSERT INTO Person <<{'FirstName' : 'Raul',
+    'LastName' : 'Lewis',
+    'DOB' : 1963-08-19T,
+    'GovId' : 'LEWISR261LL',
+    'GovIdType' : 'Driver License',
+},{'LastName' : 'Logan',
+    'DOB' : 1967-07-03T,
+    'Address' : '43 Stockert Hollow Road, Everett, WA, 98203'
+},{'LastName' : 'Pena',
+    'DOB' : 1974-02-10T,
+    'GovId' : '744 849 301',
+    'GovIdType' : 'SSN',
+    'Address' : '4058 Melrose Street, Spokane Valley, WA, 99206'
+}>>;
+```
+
+### 3.2 Query
+
+`SELECT` query expression can be used for inserting the resulting values from the query to the target table. The resulting tuples (e.g. rows) MUST have the same attribute layout as the target schema’s required attributes.
+
+#### 3.4 Examples
+
+The following example statement inserts items from `Vehicles` table to `HeavyVehicles` table. Both tables have the same layout for required attributes, therefore the INSERT statement can insert items using a SELECT query. The second INSERT statement shows that the attributes can be specified INSERT and SELECT.
+
+```SQL
+-- The following `CREATE TABLE` is arbitrary since the PartiQL DDL is yet to be
+-- defined.
+CREATE TABLE Music (
```

Comment:
I'm going to be the obnoxious person that ask when is DDL planned for? Some of the examples above (i.e. VALUES and DEFAULT) presume that there's a closed schema somewhere, but we don't have a facility in PartiQL to create that schema in the first place.

Reply:
Asking for DDL is fair, however I see that as being orthogonal to this RFC. This Spec. assumes there is a set of data-items stored in some database system that requires manipulation with PartiQL; for example, using the `INSERT` statement to manipulate data on DynamoDB. The intention for formalizing this RFC comes from a direct customer ask that's why we've prioritized. 

For DDL, as of now we haven't planned for it in short-term; would be great to have a further discussion to understand your use-case and to see how we can re-baseline our plans if needed.